client-app/ui-kit/components/molecules/rating/vc-rating.vue

```diff
@@ -1,37 +1,150 @@
 <template>
-  <div class="vc-rating">
-    <VcIcon class="vc-rating__icon" name="star" size="xs" />
-    <div class="vc-rating__text">
-      <span class="vc-rating__current-rating">{{ rating.value }}</span>
-      /5 ({{ rating.reviewCount }})
+  <div
+    :class="['vc-rating', `vc-rating--mode--${mode}`, `vc-rating--size--${size}`, { 'vc-rating--read-only': readOnly }]"
+  >
+    <template v-if="mode === 'mini'">
+      <VcIcon class="vc-rating__icon" name="whishlist" :size="size" />
+    </template>
+
+    <template v-else>
+      <VcShape
+        v-for="i in MAX_RATING"
+        :key="i"
+        mask="whishlist"
+        :bg-color="getColor(i)"
+        @click="setRating(i)"
+        @focus="handleMouseOver(i)"
+        @blur="handleMouseOver(null)"
+        @mouseover="handleMouseOver(i)"
+        @mouseleave="handleMouseOver(null)"
+      >
+        <div v-if="isHalf(i)" class="vc-rating--bg" />
+      </VcShape>
+    </template>
+
+    <div v-if="withText" class="vc-rating__text">
+      <span class="vc-rating__current-rating">{{ value }}/5</span>
+      <span v-if="reviewCount"> ({{ reviewCount }})</span>
     </div>
   </div>
 </template>
 
 <script setup lang="ts">
-import type { Rating } from "@/core/api/graphql/types";
+import { ref } from "vue";
+import { MAX_RATING } from "@/modules/customer-reviews/constants";
+
+interface IEmits {
+  (event: "setRating", value: number): void;
+}
 
 interface IProps {
-  rating: Rating;
+  mode?: "mini" | "full";
+  readOnly?: boolean;
+  reviewCount?: number;
+  size?: "xs" | "sm" | "md";
+  value: number;
+  withText?: boolean;
+}
+
+const emit = defineEmits<IEmits>();
+
+const props = withDefaults(defineProps<IProps>(), {
+  mode: "mini",
+  size: "md",
+  withText: true,
+});
+
+const selectedRating = ref<number | null>(null);
+
+const getColor = (index: number): string => {
+  const rating = selectedRating.value ?? props.value;
+  const fullStars = Math.floor(rating);
+
+  if (index <= fullStars) {
+    return "--color-primary-500";
+  }
+
+  return "--color-neutral-300";
+};
+
+const isHalf = (index: number): boolean => {
+  const rating = selectedRating.value ?? props.value;
+  const fullStars = Math.floor(rating);
+  const halfStar = rating % 1 !== 0 ? 1 : 0;
+  return Boolean(index === fullStars + 1 && halfStar);
+};
+
+function handleMouseOver(value: number | null): void {
+  if (props.readOnly) {
+    return;
+  }
+  selectedRating.value = value;
 }
 
-defineProps<IProps>();
+function setRating(value: number): void {
+  if (props.readOnly) {
+    return;
+  }
+  emit("setRating", value);
+}
 </script>
 
 <style lang="scss">
 .vc-rating {
-  @apply flex items-center gap-0.5;
+  @apply flex items-center gap-0.5 cursor-pointer;
+
+  &--size {
+    &--xs {
+      @apply text-xs;
+
+      --vc-shape-size: theme("fontSize.sm");
+
+      .vc-rating__text {
```

Comment:
Do not duplicate classes. Use sass variables, or css variables, or change layout.